### ISSUE TITLE
Incorrect HTTP status on document validation

### ISSUE THREAD

**Issue by dmoidl** [Captain-P-Goldfish/SCIM-SDK#384]
Ahoy Captain 🫡

First of all, thanks for all your effort put into this library. It has already saved us a great deal of work while implementing a SCIM service provider 👍 However, during testing our implementation we stumbled upon the issue described below.

As we were testing all possible scenarios, we tried sending a payload meant to be sent via `PUT` using the `PATCH` HTTP method. This was to simulate an erroneous input being send via `PATCH`. What we got back from the server was a generic "internal server error" response:
```
{
    "detail": "An internal error has occurred.",
    "schemas": [
        "urn:ietf:params:scim:api:messages:2.0:Error"
    ],
    "status": 500
}
```
whereas we were expecting a HTTP status of `400` with some reasonable error message.

Internally, we did get a more useful error message logged:
```
Required 'WRITE_ONLY' attribute 'urn:ietf:params:scim:api:messages:2.0:PatchOp:Operations' is missing
```
But this does not get propagated to the user, which is unfortunate as the generic "internal server error" response doesn't really convey any useful information for the user.

The corresponding stack trace shows the exception originated [here](https://github.com/Captain-P-Goldfish/SCIM-SDK/blob/c65cf1cb271f5aabc2a53f7076b2399a2b142b12/scim-sdk-server/src/main/java/de/captaingoldfish/scim/sdk/server/schemas/validation/AbstractSchemaValidator.java#L75). I checked the code and it looks like for some reason, in this case the `DocumentValidationException` is intentionally instantiated without any HTTP status 🤔 That, together with [this condition](https://github.com/Captain-P-Goldfish/SCIM-SDK/blob/master/scim-sdk-common/src/main/java/de/captaingoldfish/scim/sdk/common/response/ErrorResponse.java#L61), makes the library return the generic error response.

Now, I don't see deep enough into the implementation to be able to tell whether this is some kind of a special case where the `500` response is returned intentionally. However, as a user, I find this strange as naturally, I would expect a `DocumentValidationException` to always map to "bad request".

I'm happy to issue a PR if you confirm this is not the intended behavior 🙂

**Comment by Captain-P-Goldfish** [comment]
Hi, I am glad that the library is of help to you :-)

I was able to confirm your issue but we got indeed a special case here. This code is called twice during processing. 
1. on request which should result in a 400 status as you already pointed out
2. on response building which should result in 500 because the error would be on server side.

In this special case an abstract method for the expected response code would be appropriate. The MetaSchemaValidator and the ResponseSchemaValidator would then return 500 and the RequestSchemaValidator could return the 400 status code.
